tests/datasource/data_connector/test_runtime_data_connector.py

```diff
@@ -310,8 +309,28 @@ def test_get_batch_definition_list_from_batch_request_length_zero(
     ] = test_runtime_data_connector.get_batch_definition_list_from_batch_request(
         batch_request=batch_request
     )
+    assert len(batch_definition_list) == 1
+    # check that default value has been set
+    assert batch_definition_list[0]["data_asset_name"] == "IN_MEMORY_DATA_ASSET"
 
-    assert len(batch_definition_list) == 0
+    # test that name can be set as "my_data_asset"
+    batch_request: dict = {
+        "datasource_name": basic_datasource.name,
+        "data_connector_name": test_runtime_data_connector.name,
+        "data_asset_name": "my_data_asset",
+        "batch_data": test_df,
+        "partition_request": partition_request,
+        "limit": None,
+    }
+    batch_request: BatchRequest = BatchRequest(**batch_request)
+    batch_definition_list: List[
+        BatchDefinition
+    ] = test_runtime_data_connector.get_batch_definition_list_from_batch_request(
+        batch_request=batch_request
+    )
+    assert len(batch_definition_list) == 1
+    # check that default value has been set
+    assert batch_definition_list[0]["data_asset_name"] == "my_data_asset"
```

Comment:
I do not think that this test probes the presence of `data_asset_name` sufficiently deeply.  While it may exist in a `BatchDefinition`, is it attributed to the given `Batch`/`Partition` and can it be queried from it across multiple calls to the given data connector?  I feel that needing more time to figure this out properly was one of the main reasons why we left this unfinished during BatchPod (and had to settle for `IN_MEMORY_DATA_ASSET` as the default).

Reply:
This is a very very good question. I'll look into this